src/Umbraco.Web.UI.Client/src/packages/property-editors/checkbox-list/property-editor-ui-checkbox-list.element.ts

```diff
@@ -89,6 +93,23 @@ export class UmbPropertyEditorUICheckboxListElement
 		this.dispatchEvent(new UmbChangeEvent());
 	}
 
+	/**
+	 * Updates the checked state of all list items based on current selection.
+	 * This fixes the issue where UI doesn't update when values are set programmatically.
+	 */
+	#updateCheckedState() {
+		// Only update if we have list items loaded
+		if (this._list.length > 0) {
+			// Update state only if changes are needed
+			const updatedList = updateItemsSelectedState(this._list, this.#selection, 'checked');
+			if (updatedList !== this._list) {
+				this._list = updatedList;
```

Comment:
Because _list is a @state and you are setting the property, not just manipulating it, there should be no need to call requestUpdate(). I will double check this, but just informing you thats how Lit was suppose to work.

If its of no use I will remove it if I have the access.

Reply:
Just confirming that it was not needed, I have removed it. So nothing is required from you.